blog/_posts/2020-10-20-bsp-in-sbt.md

```diff
@@ -0,0 +1,153 @@
+
+We, at the Scala Center, had the chance to closely collaborate with [Eugene Yokota](http://eed3si9n.com/) on this proposal and we are now proud to announce that BSP is supported by [sbt 1.4.0][release-note]. In the present post we outline the motivation behind BSP in sbt and the following outcome.
+
+## Motivation
+
+The idea of BSP emerged after facing the fact that the integration of build tools inside IDEs requires a fair amount of work and maintenance effort, which is multiplied by the ever growing number of available build tools.
+
+This integration is often fragile because of the highly customizable nature of build tools. It is quite common for the users of IDEs to experience false compiler errors or out-of-sync state of the dependencies or the generated source files.
+
+By formalizing BSP, we aimed at providing a standard protocol of communication between IDEs and build tools, in which the build tool plays the role of the server that performs the operation requested by the IDE. The ultimate goal being to ease the integration on both sides while providing a better experience to the end-users.
+
+BSP is inspired by LSP, the [Language Server Protocol](https://microsoft.github.io/language-server-protocol/). The main difference being that LSP abstracts over the language whereas BSP abstracts over the build tool.
+
+![BSP clients](/resources/img/blog/bsp.png)
+
+Metals is both a language server and a build client. It provides text editors with support of the Scala language, enabling code-edition related features such as error reporting, code completion, go-to-definition, and more. But Metals itself depends on a build server to perform build-related operations, such as fetching the dependencies, invoking the compiler, running the tests or the application.
+
+IntelliJ Idea is also a build client, but it does not rely on a language server. It interacts directly with the build server to import the project and perform compilation on save. The Scala language support in IntelliJ is embedded in the Scala plugin.
+
+## What's new
+
+With the recent built-in support of BSP in sbt, it is going to be possible for Metals and IntelliJ to connect to the sbt server and communicate with it directly. This solution is an alternative to the current status quo of using Bloop as a third-party BSP server.
+
+Bloop still offers many advantages compared to sbt server. It can serve several build clients, on different projects, and run the requests concurrently. It also supports DAP, the Debug Adapter Protocol, which provides code editors with the ability to debug applications and evaluate code at runtime.
+
+Choosing Bloop or sbt as the build server depends on the project you are working on and the developer experience you are looking for. In the following paragraph we describe the main characteristics of using sbt as a build server.
```

Comment:
It would be nice to contrast the below to what you get when you use bloop as a build server.